src/crypto/public_key/elliptic/ecdh.js

```diff
@@ -87,28 +90,30 @@ async function kdf(hash_algo, X, length, param, stripLeading=false, stripTrailin
  * @async
  */
 async function genPublicEphemeralKey(curve, Q) {
-  switch (curve.name) {
+  switch (curve.type) {
     case 'curve25519': {
-      const { secretKey: d } = nacl.box.keyPair();
+      const d = await random.getRandomBytes(32);
       const { secretKey, sharedKey } = await genPrivateEphemeralKey(curve, Q, null, d);
       let { publicKey } = nacl.box.keyPair.fromSecretKey(secretKey);
       publicKey = util.concatUint8Array([new Uint8Array([0x40]), publicKey]);
       return { publicKey, sharedKey }; // Note: sharedKey is little-endian here, unlike below
     }
-    case 'p256':
-    case 'p384':
-    case 'p521': {
+    case 'web': {
       if (curve.web && util.getWebCrypto()) {
         try {
           return await webPublicEphemeralKey(curve, Q);
         } catch (err) {
           util.print_debug_error(err);
         }
       }
+      break;
+    }
+    case 'node': {
+      return nodePublicEphemeralKey(curve, Q);
     }
```

Comment:
Very nitpicky, but: the curly brackets `{}` are only necessary when you have `let/const` declarations in them; in the web and node case they're unnecessary. (Some other places too - I know they were there already, it was because of the `const result = await ...` that used to be there.)